src/rust/engine/Cargo.toml

```diff
@@ -149,7 +153,7 @@ task_executor = { path = "task_executor" }
 tempfile = "3"
 testutil_mock = { package = "mock", path = "testutil/mock" }
 time = "0.3"
-tokio = { version = "1.16", features = ["macros", "rt-multi-thread"] }
+tokio = { version = "1.21", features = ["macros", "rt", "rt-multi-thread"] }
```

Comment:
Tokio is referenced about a dozen places... I _think_ that this will still have the effect of upgrading it everywhere, but would be good to bump the other versions.

Not a blocker, but... consider doing that in another PR too?

Reply:
Will do the upgrade separately.

Reply:
Rebased on top of the Rust v1.64.0 and Tokio 1.21.1 upgrades.